sql/core/src/test/scala/org/apache/spark/sql/DynamicPartitionPruningSuite.scala

```diff
@@ -1528,6 +1528,55 @@ abstract class DynamicPartitionPruningSuiteBase
       }
     }
   }
+
+  test("SPARK-38570: Fix incorrect DynamicPartitionPruning caused by Literal") {
+    withSQLConf(SQLConf.DYNAMIC_PARTITION_PRUNING_ENABLED.key -> "true") {
+      withTable("fact1", "fact2", "dim") {
```

Comment:
We already have lots of built-in tables. Cloud we change the test query to:
```sql
SELECT f.store_id,
       f.date_id,
       s.state_province
FROM   (SELECT 4 AS store_id,
               date_id,
               product_id
        FROM   fact_sk
        WHERE  date_id >= 1300
        UNION ALL
        SELECT 5 AS store_id,
               date_id,
               product_id
        FROM   fact_stats
        WHERE  date_id <= 1000) f
       JOIN dim_store s
         ON f.store_id = s.store_id
WHERE  s.country = 'US'
```

Reply:
+1

Reply:
So embarrassing, I wanted to use these built-in tables before, but I didn't see table fact_stats.